akka-http2-support/src/main/scala/akka/http/impl/engine/http2/framing/Http2FrameRendering.scala

```diff
@@ -44,11 +59,30 @@ private[http] class Http2FrameRendering extends GraphStage[FlowShape[FrameEvent,
 
     override def onPull(): Unit = pull(frameIn)
 
+    private def splitByPayloadSize(d: DataFrame, size: Int): immutable.Seq[DataFrame] = {
+      debug(s"Splitting up too large data-frame into smaller frames due to exceeding max frame size. Length: ${d.payload.length}, max: ${size}")
+      val parts = d.payload.grouped(size) // TODO optimise, splitAt would be better
+
+      if (d.endStream) {
```

Comment:
The point is that this should never be reached right now because we all `DATA` frames we generate with `endStream = true` are empty so it shouldn't reach this method anyway. So, we don't actually have to handle this case right now.